store-cli.go

```diff
@@ -43,62 +42,70 @@ func makeURL(storeType, scope, key string) (*url.URL, error) {
 	storeURL := os.Getenv("SD_STORE_URL")
 	version := "v1"
 
-	var folder string
+	var scopeEnv string
+	switch scope {
+	case "events":
+		scopeEnv = os.Getenv("SD_EVENT_ID")
+	case "jobs":
+		scopeEnv = os.Getenv("SD_JOB_ID")
+	case "pipelines":
+		scopeEnv = os.Getenv("SD_PIPELINE_ID")
+	}
+
+	var path string
 	switch storeType {
 	case "cache":
-		folder = "caches"
-	case "logs":
+		path = "caches/" + scope + "/" + scopeEnv + "/" + key
 	case "artifacts":
-		folder = "builds"
+		path = "builds/" + os.Getenv("SD_BUILD_ID") + "-ARTIFACTS/" + key
+	case "logs":
+		path = "builds/" + os.Getenv("SD_BUILD_ID") + "-" + key
+	default:
+		path = ""
 	}
 
-	var path string
-	switch scope {
-	case "events":
-		path = "events/" + os.Getenv("SD_EVENT_ID") + "/" + key
-	case "builds":
-		path = os.Getenv("SD_BUILD_ID") + "-" + key
+	var fullpath string
+	if len(path) > 0 {
+		fullpath = fmt.Sprintf("%s/%s/%s", storeURL, version, path)
+	} else {
+		fullpath = fmt.Sprintf("%s/%s", storeURL, version)
```

Comment:
when would we have this case?

Reply:
When it's an invalid call? We don't want arbitrary URLs being built based on whatever is being passed in, I suppose

Reply:
Would returning error be better?

Reply:
Yeah working on the error/validation part. Thought I'd make it a separate PR but I'll push it to this PR as well